src/main/java/com/redhat/composer/services/ChatBotService.java

```diff
@@ -137,20 +164,63 @@ public Multi<String> chat(ChatBotRequest request) {
       });
       return multi;
     } catch (Exception e) {
-      log.error("Error in ChatBotService.chat", e);
+      Log.error("Error in ChatBotService.chat", e);
       return Multi.createFrom().failure(e);
     }
   }
 
   // TODO: Support non-streaming chat?
 
+  /**
+   * Prepare an AI service with zero or more content retrievers. Content retrievers may be from a
+   * {@link RetrieverRequest} or may be generated from provided {@link Document} that will be split, embedded,
+   * and stored in an {@link EmbeddingStore} for retrieval by the AI
+   *
+   * @param <T> the type of AI service to build
+   * @param aiServiceClass the class of AI service to build
+   * @param llm the language model to use in the AI service
+   * @param retrieverRequest an optional {@link RetrieverRequest} to provide additional knowledge to the AI
+   * @param documents an optional collection of {@link Document} in which the AI may infer knowledge
+   * @return the built AI service
+   */
+  private <T extends BaseAiService> T prepareAiService(
+      Class<T> aiServiceClass,
+      StreamingChatLanguageModel llm,
+      RetrieverRequest retrieverRequest,
+      Collection<Document> documents
+  ) {
+    AiServices<T> builder = AiServices.builder(aiServiceClass)
+        .streamingChatLanguageModel(llm);
+
+    List<ContentRetriever> retrievers = new ArrayList<>();
```

Comment:
Actually we eventually want to expand this logic out so that we can use multiple retrievers. With this class that change should be simpler to make which is nice.

Reply:
I'll enhance the prepareAiService() method signature to facilitate having multiple retrievers. It won't do anything diifferent right now, but it'll be easier to extend for multiple retrievers in the future.